### ISSUE TITLE
`VARIABLE_NAME_INCORRECT_FORMAT` false positive when a property has a backing field

### ISSUE THREAD

**Issue by 0x6675636b796f75676974687562** [saveourtool/diktat#1711]
## Describe the bug

Consider this idiomatic example [taken directly](https://kotlinlang.org/docs/properties.html#backing-properties) from [https://kotlinlang.org](kotlinlang.org):

```kotlin
package com.example

class MutableTableContainer {
    private var _table: Map<String, Int>? = null

    val table: Map<String, Int>
        get() {
            if (_table == null) {
                _table = hashMapOf()
            }
            return _table ?: throw AssertionError("Set to null by another thread")
        }
}
```

Diktat **1.2.5** warns:

```
MutableTableContainer.kt:4:17: [VARIABLE_NAME_INCORRECT_FORMAT] variable name should be in lowerCamelCase and should contain only latin (ASCII) letters or numbers and should start from lower letter: _table (diktat-ruleset:identifier-naming)
```

Probably we should make an exception for backing properties whenever there's an **explicit backing field** with the corresponding name (`xyz` &rarr; `_xyz`) and of the same type.

Requirements:

 1. the **backing field** should be `private` (or have a **lower** [visibility](https://kotlinlang.org/docs/visibility-modifiers.html) than that of the **property**);
 2. the **backing field** should have a matching name (`xyz` &rarr; `_xyz`);
 3. the **backing field** should have the same type as the **property** (either convert both types to a _simple name_, or resolve to a _FQN_ before you compare; nullable subtypes should be permitted: `Map<String, Int>` &rarr; `Map<String, Int>?`);
 4. the **backing field** should have no accessors/modifiers of its own;
 5. the **property** should have at least an accessor, or a modifier, or both.

Related:

 - [ ] #1709

## Expected behavior

No warning should have been reported.

## Observed behavior

A warning was reported.

## Steps to Reproduce

Check the above code example with _Diktat_.

## Environment information
* diktat version: **1.2.5**
* build tool (maven/gradle): any
* how is diktat run (CLI, plugin, etc.): doesn't matter
* kotlin version: **1.9**
* operating system: any
* link to a project (if your project is public):
